raiden/ui/web/src/app/components/token-input/token-input.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, Input } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { MatCheckboxChange } from '@angular/material';
+import { BigNumber } from 'bignumber.js';
+import { amountFromDecimal, amountToDecimal } from '../../utils/amount.converter';
+
+
+@Component({
+    selector: 'app-token-input',
+    templateUrl: './token-input.component.html',
+    styleUrls: ['./token-input.component.css'],
+})
+export class TokenInputComponent {
+
+    @Input() inputType: string;
+    @Input() inputDescription: string;
+
+    inputControl: FormControl = new FormControl(0);
+    checkboxControl: FormControl = new FormControl({
+        value: true,
+        disabled: true
+    });
+
+    constructor() {
+    }
+
+    private _decimals = 0;
+
+    public get decimals(): number {
+        return this._decimals;
```

Comment:
Hmm, this getter seems to overlap with `allowedDecimals`, can't you get rid of it?

Reply:
The overlap was a mistake on my side. I can see that what I did is confusing, so I will have to change it.

`decimals` is the value returned by the contract and when it's not 0 you get a hint about the fraction of the token that the integer number represents.

`allowedDecimals` should have a different naming since it represents the decimals included in the `tokenAmount` so maybe it should be something like `tokenAmountDecimals` or `selectedDecimals`. I will also change the name of the setter to make sure that it is not confusing.